### PR TITLE
Showing if event is confirmed in room view

### PR DESCRIPTION
**Pull request by quirk0o**
@nowNick @lkudyba @jtkpiotr

### CODE REVIEW

File: app/models/google_calendar/event_finder.rb [hunk 1]
```diff
@@ -38,6 +40,11 @@ def list_events(time_interval, rooms)
       all_events
     end
 
+    def include_confirmation(events)
+      confirmed = Event.confirmed.where(event_id: events.map { |e| e[:id] }).pluck(:event_id)
```
Comment (jtkpiotr) [inline app/models/google_calendar/event_finder.rb]: listed_event_ids = events.map { |e| e[:id] }
confirmed = Event.confirmed_event_ids & listed_event_ids ??

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -40,7 +40,12 @@ def index
   end
 
   def room_index
-    events = google_event_client.find_by_room(time_interval_rfc3339, params[:conference_room_id].to_i)
+    with_confirmation = (params[:confirmation] == 'true'.freeze)
```
Comment (jtkpiotr) [inline app/controllers/events_controller.rb]: I would extract it to another method

File: app/models/event.rb [hunk 3]
```diff
@@ -15,5 +15,8 @@ def self.confirm_or_create(conference_room_id, event_id)
   end
 
   scope :confirmed, -> { where(confirmed: true) }
-  scope :confirmed_event_ids, -> { confirmed.pluck(:event_id) }
```
Comment (nowNick) [inline app/models/event.rb:18]: What is behind this change?

Comment (quirk0o) [reply app/models/event.rb:18]: [Henryk] It didn't return ActiveRecord_Relation, so I thought it shouldn't be a scope.

File: app/controllers/events_controller.rb (hunk 2 above)
Comment (mjk-old) [reply app/controllers/events_controller.rb]: And I don't think this `freeze` is needed here. Am I wrong?

File: app/models/google_calendar/event_finder.rb (hunk 1 above)
Comment (mjk-old) [reply app/models/google_calendar/event_finder.rb]: What about:

``` ruby
def include_confirmation(events)
  events.each { |event| event[:confirmed] = confirmed_events_ids.include?(event[:id]) }
end

def confirmed_events_ids
  @confirmed_events_ids ||= Event.confirmed.pluck(:event_id)
end
```

File: spec/models/google_calendar/event_finder_spec.rb [hunk 4]
```diff
@@ -191,4 +191,60 @@
...
     it { is_expected.to eq [confirmed_event] }
   end
+
+  describe '#by_room' do
+    let(:event_id) { 'sampleid'.freeze }
+    let!(:room) { create(:conference_room) }
+    let(:start_time) { Time.now }
+    let(:end_time) { start_time + 1.hour }
+    let(:google_event) do
+      Google::Apis::CalendarV3::Event.new(start: Google::Apis::CalendarV3::EventDateTime.new(date_time: start_time),
```
Comment (mjk-old) [inline spec/models/google_calendar/event_finder_spec.rb]: Please, please, extract these parameters to their own lets.

File: spec/models/google_calendar/event_finder_spec.rb [hunk 5]
```diff
@@ -191,4 +191,60 @@
...
+      allow(GoogleOauth).to receive(:user_email) { 'example@com' }
+      allow(events).to receive(:items) { [event] }
+      allow(service).to receive(:batch).and_yield(service)
+      allow(service).to receive(:list_events) do |email, &block|
+        block.call [double('EventList', items: [google_event]), nil]
+      end
+    end
+
+    it 'gets events from room' do
+      expect(event_finder.by_room(time_interval, [room.id]).first[:summary]).to eq google_event.summary
```
Comment (mjk-old) [inline spec/models/google_calendar/event_finder_spec.rb]: Move these invocations to `subject`.

File: app/models/google_calendar/event_finder.rb (hunk 1 above)
Comment (jtkpiotr) [reply app/models/google_calendar/event_finder.rb]: Event.confirmed.pluck(:event_id) == Event.confirmed_event_ids

File: app/models/google_calendar/event_finder.rb [hunk 6]
```diff
@@ -20,8 +20,10 @@ def confirmed_events(time_interval)
       all_google_events.select { |event| confirmed_ids.include?(event.id) }
     end
 
-    def by_room(time_interval, conference_room_ids)
-      list_events(time_interval, rooms(conference_room_ids))
+    def by_room(time_interval, conference_room_ids, with_confirmation = false)
```
Comment (jtkpiotr) [inline app/models/google_calendar/event_finder.rb:23]: I believe that using boolean flag to control method's behaviour is not recommended. Probably it's better to create another method that specializes in particular task

**Pull request merged**